modules/core/src/main/java/org/apache/ignite/IgniteCache.java

```diff
@@ -639,6 +639,10 @@
      * <p>
      * Please refer to documentation for {@link CacheAtomicityMode#ATOMIC} for information on
      * system behavior in crash scenarios for atomic caches.
+     * <p>
```

Comment:
Ilya, how about this version? I would use it here and in similar places below.

_Note, always pass the keys in ordered maps like TreeMap to avoid distributed deadlocks. Ignite always locks/accesses the keys in a way they ordered within a data structure. Don't use unordered maps like HashMap to avoid distributed deadlocks that happen if this or similar method is called by multiple threads/applications and keys are passed in an unordered map._

Reply:
I think this is a good passage for dedicated docs page, but not for copying over and over across javadoc. I will check my message to see if it can be improved.